src/parser/ascii.rs

```diff
@@ -0,0 +1,212 @@
+use btoi::btou;
+use nom::{
+    branch::alt,
+    bytes::streaming::{tag, take, take_until, take_while1, take_while_m_n},
+    character::{is_digit, streaming::crlf},
+    combinator::{map, map_res, opt, value},
+    multi::fold_many0,
+    sequence::{preceded, terminated, tuple},
+    IResult,
+};
+
+use super::{ErrorKind, Response, Status, Value};
+
+fn _parse_ascii_status(buf: &[u8]) -> IResult<&[u8], Response> {
```

Comment:
any reason to prefix the method with an underscore?

Reply:
yes, public version is alternated between status and error

```
pub(crate) fn parse_ascii_status(buf: &[u8]) -> IResult<&[u8], Response> {
    alt((_parse_ascii_status, parse_ascii_error))(buf)
}

```